wiki/configuration/getting-started/en.md

```diff
@@ -0,0 +1,153 @@
+- We've changed the type from `String` to `Integer`. Now, why `Integer` instead of `int` like we would use for a normal number field? `int` is a *primitive type*, which means that it isn't actually a class! Since generics can only take classes, Java provides class versions of each primitive. `boolean` becomes `Boolean`, `float` becomes `Float`, `double` becomes `Double`, `char` becomes `Character`, etc etc.
+
+Something incredibly important to remember is that you can't just send any class into `TrackedValue`'s generics: Quilt Config has to know how to serialize it. By default, all [primitive types](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html) are serializable, as well as the `String` class. We'll cover how to explain to Quilt Config how to serialize a class later, but first: more fun with annotations!
+
+Anyway. New value!
+
+`src/main/com/example/example_mod/ExampleModConfig`:
+
+```java
+public class ExampleModConfig extends ReflectiveConfig {
+	// ...
+	@Comment("Whether to print the message at all.")
+	public final TrackedValue<Boolean> print = this.value(true);
+}
+```
+
+Here we simply leverage the `Comment` annotation to add a comment explaining exactly what the config field does. This will be saved by Quilt Config in the config file, as long as the format you use supports comments. Neat!
+
+Finally, we're going to use one more annotation to tie it all together. By default, Quilt Config serializes to [TOML](https://toml.io/en/) which has a couple standards we're not following. In Java, `public final` fields are named using `lowerCamelCase`, like our `timesToPrint` field. But, in a TOML file, names should use `snake_case`, meaning we should have named the field `times_to_print`. To match both conventions, we can use the `SerializedName` annotation!
+
+`src/main/com/example/example_mod/ExampleModConfig`:
+
+```java
+public class ExampleModConfig extends ReflectiveConfig {
+	// ...
+	@IntegerRange(min = 0, max = 10)
+    @SerializedName("times_to_print")
+	public final TrackedValue<Integer> timesToPrint = this.value(1);
+	// ...
+}
+```
+
+Problem solved! Now it'll follow the Java conventions in the Java code, and the TOML conventions in the saved TOML file. We can apply `SerializedName` annotations to every multi-word config field to solve our problems everywhere.
+
+## Maps and lists
+
+Two of the most common data structures used in Java programming are `Map` and `List`. Quilt Config provides convenient serializable versions of both of these in the for of `ValueMap` and `ValueList`!
+
+Starting with maps, `ReflectiveConfig` provides us a handy `map` method to help us easily make a default value.
+
+`src/main/com/example/example_mod/ExampleModConfig`:
+
+```java
+public class ExampleModConfig extends ReflectiveConfig {
+	// ...
+	@Comment("When a key in this map shows up in the message, it'll be replaced with the corresponding value.")
+	public final TrackedValue<ValueMap<String>> replacements = this.map("")
```

Comment:
Off topic for this PR but i wonder if this could be `this.<String>map()`

Reply:
someone who knows java better than me definitely needs to take a look at qconf

Reply:
the argument is basically a refied type to get around type erasure. it could probably be a .class instead but to work around it fully you'd need TypeTools (not happening), bytecode analysis of the class (not happening), or to rewrite the library in Kotlin (not happening)

(unless there's something I'm missing/misremembering, ofc)

Reply:
why are we ruling out bytecode hacks?

Reply:
Basically, reified types aren't guaranteed to actually _be_ there. It's basically a JVM implementation detail - and there's lots of cases where they won't be there, like if it were `this.<T>map()` for some type parameter or whatever - that would compile fine, but your runtime reified type finder would fail terribly. Furthermore, not all JVM languages add them in bytecode the same way or at all - you'd almost certainly mess up groovy support if you did this sort of thing.

Reply:
(this is also, imo, why reflective serialization in general is a bad approach, cause you run into this issue pretty quickly, but I also understand the appeal as it's way faster to write)

Reply:
@lukebemish thanks for the insight!
just wondering -- do you know an alternative to reflective serialization that we could consider?